src/fennel/binary.fnl

```diff
@@ -235,12 +235,14 @@ int main(int argc, char *argv[]) {
     (os.exit 0)))
 
 (fn native-path? [path]
-  (match (path:match "%.(%a+)$")
-    :a path
-    :o path
-    :so path
-    :dylib path
-    _ false))
+  (let [(extension version-extension) (path:match "%.(%a+)(%.?%d*)$")]
+    (if (and version-extension (not (version-extension:match "%.%d+"))) false
```

Comment:
Actually, I think this needs to check for `version-extension` being an empty string here as well.